### DISCUSSION TITLE
Macro's not Syncing between desktop and mobile

### DISCUSSION THREAD

**Discussion by scottrlarson** [chhoumann/quickadd#723] [General]
Im having some trouble getting macros to sync across devices. I tried reinstalling on all devices but for some reason only one of my devices has the macros and obsidian sync wont sync the macros over. Macros are suppose to sync correct?

**Comment by shaqmughal** [comment]
I’m also have the exact same issue. Got everything setup on my laptop. I don’t see any of the macros available on my iPad.

**Reply by shaqmughal** [reply]
Seems like this might be isolated to iOS (in my experience). I have been traveling working on one computer (mac). I updated a different computer (mac) and my macros synced fine and are available. I tried again on my iPad, no luck. Feels like maybe iOS devices are not syncing or running QuickAdd properly?

**Comment by chhoumann** [comment]
Sorry for the very late reply.

Macros live in QuickAdd's settings file, `.obsidian/plugins/quickadd/data.json` - so whether they sync is entirely up to Obsidian Sync's vault configuration settings, and those are configured **per device**. That is almost certainly why your Macs agree with each other while the iPad does not: the toggles are off on the iPad itself.

On each device, go to Settings -> Sync -> Vault configuration sync and enable **Installed community plugin list** (and **Active community plugin list**). They are off by default, and plugin settings like QuickAdd's choices and macros only transfer when they are on.

Two more things worth checking:

- If any macro runs a user script (a `.js` file in your vault), also enable **Sync all other types** under the synced file types - `.js` files are not among the types Obsidian Sync carries by default, so the macro would arrive without its script.
- After the sync finishes on the iPad, restart Obsidian (or toggle QuickAdd off and on) so the plugin reloads its settings file.

If the toggles are on everywhere and the macros still do not appear on the iPad, that would be worth a fresh issue.
